test_data/rules/no_leading_underscores_for_local_identifiers.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `dart test -N no_leading_underscores_for_local_identifiers`
+
+var _foo = 0; // OK
+const _foo1 = 1; // OK
+final _foo2 = 2; // OK
+
+void fn() {
+  var _foo = 0; // LINT
+  const _foo1 = 1; // LINT
+  final _foo2 = 2; // LINT
+  var foo_value = 0; // OK
+  var foo__value = 0; // OK
+  var foo__value_ = 0; // OK
+}
+
+void fn2(_param1) => null; // LINT
+
+void fn3(param) => null; // OK
+
+void fn4(_) => null; // OK
+
+void fn5(param_value) => null; // OK
+
+class TestClass {
+  var _foo = 0; // OK
+  static const _foo1 = 1; // OK
+  final _foo2 = 2; // OK
+
+  void foo() {
+    var _foo = 0; // LINT
+    const _foo1 = 1; // LINT
+    final _foo2 = 2; // LINT
+    var foo_value = 0; // OK
+    var foo__value = 0; // OK
+    var foo__value_ = 0; // OK
+
+    for(var _x in [1,2,3]) {} // LINT
+    for(var x in [1,2,3]) {} // OK
+
+    [1,2,3].forEach((_x) => fn()); // LINT
+    [1,2,3].forEach((x) => fn()); // OK
+    [1,2,3].forEach((_) => fn()); // OK
+
+    try {}
+    catch(_error) {} // LINT
+
+    try {}
+    catch(error) { // OK
+    }
+
+    try {}
+    catch(error, _stackTrace) {} // LINT
+
+    try {}
+    catch(error, stackTrace) { // OK
+    }
+
+    for (var _e in ['']) { // LINT
+    }
+
+    void bar(var _baz) { } // LINT
+  }
+
+  void foo1(_param) => null; // LINT
+
+  void foo2(param) => null; // OK
+
+  void foo3(_) => null; // OK
+
+  void foo4(param_value) => null; // OK
+
+  void foo5(param, [_positional]) => null; // LINT
+
+  void foo6(param, [positional]) => null; // OK
```

Comment:
I don't see any tests involving
- named parameters (either required or optional),
- function typed formal parameters,
- function type defs, or
- `for` elements in collection literals.